### PR TITLE
Feat/allow search on saved features (Issue #5)

### PR DESCRIPTION
**Pull request by jricardo27**
Implements a search filter for the Points of Interest (POI) list within the SavedFeaturesDrawer.

Key changes:

- Added a TextField to the SavedFeaturesDrawer for search input.
- Implemented state management for the search query.
- Introduced filtering logic that searches POI names and descriptions (case-insensitive).
- Refactored the filtering logic into a separate utility function filterFeatures in filterUtils.ts for better code organization and testability.
- Added comprehensive unit tests for the filterFeatures function to ensure correctness across various scenarios, including partial matches, case-insensitivity, and handling of missing properties.

### CODE REVIEW

File: src/components/SavedFeaturesDrawer/SavedFeaturesDrawer.tsx [hunk 1]
```diff
@@ -49,12 +51,40 @@ const SavedFeaturesDrawer: React.FC<SavedFeaturesDrawerProps> = ({ drawerOpen, o
...
+      if (!item.feature.properties) {
+        return false
+      }
+
+      const feature = item.feature // Access the feature for filtering
+      const nameMatch = feature.properties?.name?.toLowerCase().includes(query)
+      const descriptionMatch = feature.properties?.description?.toLowerCase().includes(query)
+      return Boolean(nameMatch || descriptionMatch)
+    })
+  }, [itemsWithOriginalIndex, searchQuery])
```
[the remark this answers is not in the document]
Comment (jricardo27) [reply src/components/SavedFeaturesDrawer/SavedFeaturesDrawer.tsx]: Fixed.

File: src/components/SavedFeaturesDrawer/filterFeatures.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return items
+  }
+  const query = searchQuery.toLowerCase()
+  return items.filter((item) => {
+    if (!item.feature?.properties) {
+      return false
+    }
+
+    const nameMatch = item.feature.properties.name?.toLowerCase().includes(query) ?? false
+    const descriptionMatch = item.feature.properties.description?.toLowerCase().includes(query) ?? false
```
[the remark this answers is not in the document]
Comment (jricardo27) [reply src/components/SavedFeaturesDrawer/filterFeatures.ts:17]: Not needed. Added extra tests and null case is covered.
